src/coreComponents/mesh/generators/VTKFaceBlockUtilities.cpp

```diff
@@ -14,623 +14,621 @@
  */
-class FaceData
+class ElementToFace
 {
 public:
   /**
-   * @brief Builds from the vtk input.
-   * @param fieldData The vtk field data.
-   * @param faceBlockName The face block name to search form in the @p fieldData.
+   * @brief Builds from the cell blocks.
+   * @param[in] cellBlocks The cell blocks group.
    */
-  FaceData( vtkFieldData * fieldData,
-            string const & faceBlockName )
+  explicit ElementToFace( dataRepository::Group const & cellBlocks )
+    : m_localToGlobalMaps( cellBlocks.numSubGroups() ),
+    m_elemToFacesMaps( cellBlocks.numSubGroups() )
   {
-    // First, formatting the duplicated nodes' information.
+    // In order not to copy all the mappings into a main bigger mapping,
+    // we base our implementation on the fact that for practical cases,
+    // cell blocks will contain multiple continuous ranges of cell indices.
+    // Storing those ranges and relying on the fact that numbering is continuous within each range,
+    // will save memory at the cost of look ups.
+    // Note that it's not required that the whole cell block contains a continuous.
+    // It can contain multiple chunks. Chucks of size one are supported.
+    localIndex const numCellBlocks = cellBlocks.numSubGroups();
+    for( int i = 0; i < numCellBlocks; ++i )
     {
-      string const DUPLICATED_POINTS_INFO = "duplicated_points_info";
-      vtkDataArray * duplicatedNodes = fieldData->GetArray( DUPLICATED_POINTS_INFO.c_str() );
-      GEOSX_ERROR_IF( duplicatedNodes == nullptr, "Mandatory VTK field data array \"" + DUPLICATED_POINTS_INFO + "\" not found." );
-
-      // vtk will provide the field as a (numComponents, numTuples) table.
-      // numTuples is actually the number of duplicated nodes too.
-      // But numComponents is just the maximum number of multiplied nodes at one location.
-      // For a T shaped fracture, the nodes at the T intersection will be tripled.
-      // But all the other nodes will only get doubled.
-      // Nevertheless, numComponents will be equal 3 for all the nodes.
-      // The dummy slot will be filled with negative dummy values.
-      int const numComponents = duplicatedNodes->GetNumberOfComponents();
-      int const numDuplicatedLocations = duplicatedNodes->GetNumberOfTuples();
-      for( int i = 0; i < numDuplicatedLocations; ++i )
+      CellBlock const & cb = cellBlocks.getGroup< CellBlock >( i );
+      m_localToGlobalMaps[i] = cb.localToGlobalMapConstView();
+      m_elemToFacesMaps[i] = cb.getElemToFacesConstView();
+    }
+
+    for( int i = 0; i < numCellBlocks; ++i )
+    {
+      arrayView1d< globalIndex const > const l2g = m_localToGlobalMaps[i];
+      int const size = l2g.size();
+      for( int j = 0, start = 0; j < size - 1; ++j )
       {
-        std::vector< double > rawTuple( numComponents );
-        duplicatedNodes->GetTuple( i, rawTuple.data() );
-        std::set< int > nodes;
-        for( double const & v: rawTuple )
+        if( l2g[j] + 1 != l2g[j + 1] )
```

Comment:
can you write a brief comment with the meaning of this if conditions. I am a bit lost.